cmudb/exploration/exploratory_daemon.py

```diff
@@ -0,0 +1,132 @@
+                         default=REPLICA_VOLUME_POOL)
+    # Docker args
+    aparser.add_argument("--docker-volume-directory", help="directory path of the docker volume directory",
+                         default=DOCKER_VOLUME_DIR)
+    args = vars(aparser.parse_args())
+
+    run_daemon(args["postgres_replica_port"], args["postgres_exploratory_port"], args["zfs_volume_pool"],
+               args["zfs_replica_pool_name"], args["docker_volume_directory"])
+
+
+def run_daemon(replica_port: int, exploratory_port: int, zfs_volume_pool: str, zfs_replica_pool: str,
+               docker_volume_dir: str):
+    """
+    Run exploratory daemon
+    Parameters
+    ----------
+    replica_port
+        port that replica instance is reachable from
+    exploratory_port
+        port that exploratory instance will be reachable from
+    zfs_volume_pool
+        name of zfs pool used to store docker volumes
+    zfs_replica_pool
+        relative name of zfs pool used to store postgres replica data
+    docker_volume_dir
+        directory path that docker uses for volumes
+    """
+    setup_docker_env(docker_volume_dir)
+    destroy_exploratory_data_cow(zfs_volume_pool, zfs_replica_pool)
+    # Make sure that container doesn't reuse machine's IP address
+    execute_sys_command("sudo docker network create --driver=bridge --subnet 172.19.253.0/30 tombstone")
+    exploratory_docker_proc, valid = spin_up_exploratory_instance(replica_port, exploratory_port, zfs_volume_pool,
+                                                                  zfs_replica_pool, docker_volume_dir)
+    if valid:
+        print(execute_sql("CREATE TABLE foo(a int);", EXPLORATION_PORT))
+        print(execute_sql("INSERT INTO foo VALUES (42), (666);", EXPLORATION_PORT))
+        print(execute_sql("SELECT * FROM foo;", EXPLORATION_PORT))
+    else:
+        print("Failed to start exploratory instance")
+    spin_down_exploratory_instance(exploratory_docker_proc, zfs_volume_pool, zfs_replica_pool, docker_volume_dir)
+
+
+def spin_up_exploratory_instance(replica_port: int, exploratory_port: int, zfs_volume_pool: str, zfs_replica_pool: str,
+                                 docker_volume_dir: str) -> Tuple[subprocess.Popen, bool]:
+    """
+    Start exploratory instance
+    Parameters
+    ----------
+    replica_port
+        port that replica instance is reachable from
+    exploratory_port
+        port that exploratory instance will be reachable from
+    zfs_volume_pool
+        name of zfs pool used to store docker volumes
+    zfs_replica_pool
+        relative name of zfs pool used to store postgres replica data
+    docker_volume_dir
+        directory path that docker uses for volumes
+    Returns
+    -------
+    exploratory_instance
+        docker process that is running exploratory instance
+    valid
+        True if the container started successfully, False otherwise
+    """
+    print("Taking checkpoint in replica")
+    checkpoint(replica_port)
```

Comment:
@lmwnshn I think honestly we should remove this. I'm not really sure how much we gain by calling a checkpoint and it ends up being really expensive a lot of the time.